render.go

```diff
@@ -389,14 +397,15 @@ func (r *Render) Data(w io.Writer, status int, v []byte) error {
 
 // HTML builds up the response from the specified template and bindings.
 func (r *Render) HTML(w io.Writer, status int, name string, binding interface{}, htmlOpt ...HTMLOptions) error {
-	r.templatesLk.Lock()
-	defer r.templatesLk.Unlock()
 
 	// If we are in development mode, recompile the templates on every HTML request.
 	if r.opt.IsDevelopment {
 		r.compileTemplates()
 	}
 
+	r.templatesLk.RLock()
```

Comment:
Is this lock/unlock necessary? Since all other locks just in compileTemplates ?

Reply:
Yes because this is using the templates. We can't replace the templates until they're not being used by anyone else.